goal_src/jak2/levels/dig/dig-obs.gc

```diff
@@ -7,3 +7,906 @@
+          (the-as string #f)
+          (the-as (function gui-connection symbol) #f)
+          (the-as process #f)
+          )
+        )
+      (set! (-> self state-time) (-> self clock frame-counter))
+      (until (>= (- (-> self clock frame-counter) (-> self state-time)) (seconds 0.5))
+        (suspend)
+        )
+      (let ((f30-0
+              (+ (-> (&-> *dig-log-heights* 0 data (- (-> self total-buttons) (-> self pressed-count))) 0) (-> self base-y))
+              )
+            (s5-1 (-> self clock frame-counter))
+            )
+        (until (>= (-> self root trans y) f30-0)
+          (set! (-> self root trans y)
+                (seek-with-smooth (-> self root trans y) f30-0 (* 4096.0 (-> self clock seconds-per-frame)) 0.25 40.96)
+                )
+          (when (>= (-> self clock frame-counter) s5-1)
+            (activate!
+              *camera-smush-control*
+              (rand-vu-float-range 614.4 1433.6)
+              45
+              75
+              1.0
+              0.95
+              (-> *display* camera-clock)
+              )
+            (set! s5-1 (+ (-> self clock frame-counter) (rand-vu-int-range (seconds 0.05) (seconds 0.15))))
+            )
+          (let ((a1-10 (new 'stack-no-clear 'vector)))
+            (set! (-> a1-10 quad) (-> self root trans quad))
+            (set! (-> a1-10 y) (+ 59592.703 (-> self base-y)))
+            (spawn (-> self part) a1-10)
+            )
+          (suspend)
+          )
+        )
+      (set! (-> self state-time) (-> self clock frame-counter))
+      (until (>= (- (-> self clock frame-counter) (-> self state-time)) (seconds 1))
+        (suspend)
+        )
+      (if (= (-> gp-0 cam-ret-mode) 'instant)
+          (persist-with-delay *setting-control* 'interp-time (seconds 0.05) 'interp-time 'abs 0.0 0)
+          )
+      (set-setting! 'string-startup-vector 'abs (-> gp-0 cam-ret-dir) 0)
+      (remove-setting! 'entity-name)
+      (when (!= (-> gp-0 cam-ret-mode) 'instant)
+        (set! (-> self state-time) (-> self clock frame-counter))
+        (until (>= (- (-> self clock frame-counter) (-> self state-time)) (seconds 1))
+          (suspend)
+          )
+        )
+      )
+    (set! (-> self state-time) (-> self clock frame-counter))
+    (until (>= (- (-> self clock frame-counter) (-> self state-time)) (seconds 0.1))
+      (suspend)
+      )
+    (remove-setting! 'string-startup-vector)
+    (until (process-release? *target*)
+      (suspend)
+      )
+    (set! (-> self state-time) (-> self clock frame-counter))
+    (until (>= (- (-> self clock frame-counter) (-> self state-time)) (seconds 4))
+      (suspend)
+      )
+    (go-virtual idle)
+    (none)
+    )
+  :post (the-as (function none :behavior dig-log) transform-post)
+  )
+
+;; WARN: Return type mismatch object vs none.
+(defmethod init-from-entity! dig-log ((obj dig-log) (arg0 entity-actor))
+  (local-vars (r0-0 uint128) (a0-11 object) (a1-8 uint128) (sv-16 int))
```

Comment:
sv-16 needs to be a res-tag here, otherwise it'll corrupt stack memory. I'll merge and fix it